### PR TITLE
chore: Add support for label/domain hint in long-form DIDs

### LINKED ISSUE #584: Add support for label/domain hint in long-form DIDs
If label/domain is specified for document handler create an equivalent ID with "label:domain" hint for long-form DIDs.
Also, provide label as document handler option.

### PR DESCRIPTION
**Pull request by sandrask**
If label/domain is specified for document handler create an equivalent ID with "label:domain" hint for long-form DIDs.
Also, provide label as document handler option.

Closes #584

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: pkg/dochandler/handler.go [hunk 1]
```diff
@@ -157,30 +165,44 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
...
+
+	return pv.DocumentTransformer().TransformDocument(rm, ti)
+}
+
+func (r *DocumentHandler) getTransformationInfoForUnpublished(suffix string, createRequestJCS string) protocol.TransformationInfo {
 	ti := make(protocol.TransformationInfo)
 	ti[document.PublishedProperty] = false
 
-	id := r.namespace + docutil.NamespaceDelimiter + op.UniqueSuffix
+	id := fmt.Sprintf("%s:%s", r.namespace, suffix)
```
Comment (troyronda) [inline pkg/dochandler/handler.go:177]: what about docutil.NamespaceDelimiter ?

File: pkg/dochandler/handler.go [hunk 2]
```diff
@@ -157,30 +165,44 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
...
+
+	// For interim/unpublished documents we should set optional label if specified.
+	if r.label != "" {
+		id = fmt.Sprintf("%s:%s:%s", r.namespace, r.label, suffix)
```
Comment (troyronda) [inline pkg/dochandler/handler.go:181]: what about docutil.NamespaceDelimiter

File: pkg/dochandler/handler.go [hunk 3]
```diff
@@ -157,30 +165,44 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
...
-			"interim" + docutil.NamespaceDelimiter + r.domain + docutil.NamespaceDelimiter + op.UniqueSuffix
+	if createRequestJCS != "" {
+		// we should always set short form equivalent id for long form resolution
+		equivalentIDs = append(equivalentIDs, id)
+	}
 
-		ti[document.EquivalentIDProperty] = []string{equivalentID}
+	// Also, if optional domain is specified, we should set equivalent id with domain hint
+	if r.label != "" && r.domain != "" {
+		equivalentID := fmt.Sprintf("%s:%s:%s:%s", r.namespace, r.label, r.domain, suffix)
```
Comment (troyronda) [inline pkg/dochandler/handler.go:193]: what about docutil.NamespaceDelimiter

File: pkg/dochandler/handler.go [hunk 4]
```diff
@@ -157,30 +165,44 @@ func (r *DocumentHandler) getCreateResponse(op *operation.Operation, pv protocol
...
 
-	ti[document.IDProperty] = id
+	if len(equivalentIDs) > 0 {
+		ti[document.EquivalentIDProperty] = equivalentIDs
+	}
 
-	return pv.DocumentTransformer().TransformDocument(rm, ti)
-}
+	if createRequestJCS != "" {
+		id = fmt.Sprintf("%s:%s", id, createRequestJCS)
```
Comment (troyronda) [inline pkg/dochandler/handler.go:202]: what about docutil.NamespaceDelimiter

**Review by troyronda — APPROVED** [review]

File: pkg/dochandler/handler.go (hunk 3 above)
Comment (sandrask) [reply pkg/dochandler/handler.go:193]: I find it easier to follow with Sprintf and it is no longer just namespace delimiter but just delimiter for different parts of did

**Review by bstasyszyn — APPROVED** [review]

**Pull request merged**